packages/mobile/src/screens/favorites-screen/TracksTab.tsx

```diff
@@ -0,0 +1,197 @@
+} from 'audius-client/src/common/store/pages/saved-page/selectors'
+import {
+  saveTrack,
+  unsaveTrack
+} from 'audius-client/src/common/store/social/tracks/actions'
+import { View } from 'react-native'
+import { shallowEqual, useSelector } from 'react-redux'
+
+import { Tile, VirtualizedScrollView } from 'app/components/core'
+import LoadingSpinner from 'app/components/loading-spinner'
+import { TrackList } from 'app/components/track-list'
+import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
+import { getPlaying, getPlayingUid } from 'app/store/audio/selectors'
+import { makeStyles } from 'app/styles'
+import { make, track } from 'app/utils/analytics'
+
+import { EmptyTab } from './EmptyTab'
+import { FilterInput } from './FilterInput'
+
+const messages = {
+  emptyTabText: "You haven't favorited any tracks yet.",
+  inputPlaceholder: 'Filter Tracks'
+}
+
+const useStyles = makeStyles(({ palette, spacing }) => ({
+  container: {
+    marginVertical: spacing(4),
+    marginHorizontal: spacing(3),
+    // TODO: Need to fix this spacing issue on the bottom of pages to account for the bottom bar and now playlist drawer
+    marginBottom: 240,
+    borderRadius: 6
+  },
+  trackListContainer: {
+    backgroundColor: palette.white,
+    borderRadius: 6,
+    overflow: 'hidden'
+  },
+  spinnerContainer: {
+    flexDirection: 'row',
+    justifyContent: 'center',
+    marginVertical: 48
+  }
+}))
+
+export const TracksTab = ({ navigation }) => {
+  const dispatchWeb = useDispatchWeb()
+  const styles = useStyles()
+  const [filterValue, setFilterValue] = useState('')
+  const isPlaying = useSelector(getPlaying)
+  const playingUid = useSelector(getPlayingUid)
+  const savedTracksStatus = useSelectorWeb(getSavedTracksStatus)
+  const savedTracks = useSelectorWeb(getSavedTracksLineup)
+
+  const trackIds = savedTracks.entries.map(e => e.id)
+  const tracks = useSelectorWeb(
+    (state: CommonState) => getTracks(state, { ids: trackIds }),
+    shallowEqual
+  ) as { [id: number]: Track }
+
+  const creatorIds = Object.values(tracks).map(t => t.owner_id)
+  const artists = useSelectorWeb(
+    (state: CommonState) => getUsers(state, { ids: creatorIds }),
+    shallowEqual
+  ) as { [id: number]: User }
+
+  const tracksWithUsers = Object.values(tracks).map(track => ({
+    ...track,
+    uid: savedTracks.entries.find(t => t.id === track.track_id).uid,
+    user: artists[track.owner_id]
+  }))
+
+  const isQueued = () => {
+    return tracksWithUsers.some((track: any) => playingUid === track.uid)
+  }
+
+  const queuedAndPlaying = isPlaying && isQueued
+
+  const matchesFilter = (track: any) => {
+    const matchValue = filterValue.toLowerCase()
+    return (
+      track.trackTitle.toLowerCase().indexOf(matchValue) > -1 ||
+      track.artistName.toLowerCase().indexOf(matchValue) > -1
+    )
+  }
+
+  const trackList = tracksWithUsers
+    .map(track => ({
+      isLoading: false,
```

Comment:
ah i see, i think this is fine, but i wonder if it makes sense to not have to rename all the props and instead just pass down the track unmodified to those inner components? it could simplify their api too cause they won't have like 20 props, and instead just have `track`, and like `isLoading` or something?

we can def handle this later though